pkg/api/validate/dynamic/quota.go

```diff
@@ -11,115 +11,65 @@ import (
 	"github.com/Azure/ARO-RP/pkg/api"
 )
 
-func addRequiredResources(requiredResources map[string]int, vmSize api.VMSize, count int) error {
+type azureVM struct {
+	Size      api.VMSize
+	CoreCount int
+	Family    string
+}
+
+func vmFromSize(vmType api.VMSize) (azureVM, bool) {
+	vm, ok := _vmTypesMap[vmType]
+	return vm, ok
```

Comment:
This function adds no value beyond what the map lookup does and adds unnecessary complexity, reduces readability, and adds overhead. We should change `var _vmTypesMap` to `var vmTypesMap` and replace both
```sh
vm, ok := vmFromSize(oc.Properties.MasterProfile.VMSize)
```
with
```sh
vm, ok := vmTypesMap[vmType]
```

Reply:
Yeah I thought that would make it more obvious that it was to be considered as a const, but it hasn't had the effect I was hoping for, removing it :)